design/pool-configuration.md

```diff
@@ -0,0 +1,390 @@
+
+Having an IP with no nodes advertising it is a risk. Some scenarios are:
+
+* Traffic Policy Local with pods not running on the same nodes selected by the selector
+* Node BGP peer selector with no intersection with the address selector
+
+It should be possible to detect non overlapping configurations, and raise an event when they happen. The metallb operator introduces a Validation WebHook
+that can be used to prevent those configurations to be created in first instance.
+
+### Conflicts with discovery
+
+The peer discovery feature currently being implemented in [this PR](https://github.com/metallb/metallb/pull/593) will introduce a way to
+fetch peers from the node's annotations.
+
+In order to be able to select them, the current proposed structure
+
+```yaml
+from-annotations:
+      - my-asn: example.com/local-asn
+        peer-asn: example.com/remote-asn
+        peer-address: example.com/address
+```
+
+will need to be extended with a name field (as much as the `peers` substructure) so it can be selected.
+
+### Conflicts with the FRR implementation
+
+All the BGP changes will happen from a "behavioural" point of view. Where needed, the changes will reflected in both the native and the new implementation.
+
+### Not covering all the angles
+
+The risk of not covering all possible cases / permutations is still there. For example, the proposed solution leverages pool customization to provide a
+way to set per service specific properties (i.e. define an AddressPool with bgp-communities and associate it to a single service), while a user may want to
+set custom properties only on one single service.
+
+That aside, this proposal covers the majority of the issues that were raised by the users (multiple times), and it won't prevent to add a finer level of configuration
+if ever comes the need.
+
+## Design Details
+
+### Node Selector
+
+Extending the address-pool object with an optional node selector field is straightforward:
+
+A configuration would look like:
+
+```yaml
+    peers:
+    - peer-address: 10.0.0.1
+      peer-asn: 64501
+      my-asn: 64500
+      name: foo
+    - peer-address: 10.0.0.1
+      peer-asn: 64501
+      my-asn: 64500
+      name: bar
+    address-pools:
+    - name: default
+      protocol: bgp
+      addresses:
+      - 198.51.100.0/24
+      nodeSelector:
+        matchLabels:
+          network: first
+          network: second
+```
+
+#### Implementation
+
+The [current implementation](https://github.com/metallb/metallb/blob/312b03cd3065687f25274486cd3ff5c79d6f6068/speaker/layer2_controller.go#L45) announces L2 pools only from those nodes
+with a running speaker and an active endpoint for the service.
+
+In order to be able to announce L2 from a subset of nodes, this logic must be changed, checking for active endpoints only if the TrafficPolicy is local (which is handled by the [BGP implementation](https://github.com/metallb/metallb/blob/5d236989e61683daf1eaad00777b81ce7dd98c5d/speaker/bgp_controller.go#L153)).
```

Comment:
checking for active endpoints only if the TrafficPolicy is local was done:
https://github.com/metallb/metallb/blob/36166169e7c804bd727cdfe19df6133b069c4b1b/speaker/layer2_controller.go#L100

Reply:
Yep, I did it after redacting this proposal. Removing it.